draft-ietf-quic-transport.md

```diff
@@ -1465,35 +1470,220 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+
+The server need not retransmit its PATH_RESPONSE, since it is the client's
+responsibility to send more PATH_CHALLENGE frames when a response is not
+received for a prior one.
+
+Servers MUST ignore a PATH_CHALLENGE frame from a client that is carried in a
+packet smaller than 1200 octets in size.
+
+
+#### Validating an Unvalidated Client Address {#migration-validate}
+
+Before sending any other packets to a client address that is not yet considered
+valid, the server MUST send a PATH_CHALLENGE frame with its own validation data
+to this address to verify the client's ownership of it. The server's
+PATH_CHALLENGE MAY be bundled with other frames and does not require additional
+padding.
+
+When the server receives a PATH_RESPONSE from a client and verifies that the
+included data matches that sent by the server in a previous PATH_CHALLENGE to
+the client, the server considers the client address to which the corresponding
+PATH_CHALLENGE was sent as valid.
+
+A server MAY send additional PATH_CHALLENGE frames to handle packet loss,
+subject to the congestion controller's limits.
+
+A PATH_CHALLENGE frame used for validation MUST contain at least 12 randomly
+generated octets {{?RFC4086}}, making the payload sufficiently hard to guess.
+
+The server MUST use fresh random data in every PATH_CHALLENGE frame so that it
+can associate the client's response with the causative PATH_CHALLENGE.  Not
+using fresh random data risks exposing the server to an attacker defeating
+validation by sending a spoofed PATH_RESPONSE from a victim's address.
+
+Until a client address is deemed valid, the server MUST limit the rate at
+which it sends data to this address. The server SHOULD NOT send more than an
+initial window's worth of bytes (as defined in {{QUIC-RECOVERY}}}) per estimated
+roundtrip period.  In the absence of this limit, the server risks being used for
+a denial of service attack against an unsuspecting victim.
+
+When validation of an address fails, the server SHOULD revert back to using the
+last validated client address, absent which the server MUST close the connection
+silently, and discard any further packets received from the client for this
+connection.
+
+A server MAY skip address validation for a client address that it considers to
+be already valid.
+
+After validating a client address, the server MAY send an updated address
+validation token to the client ({{address-validation}}).
+
+
+#### Committing to a New Address
+
+When a server receives a packet containing a frame other than PATH_CHALLENGE,
+PATH_RESPONSE, or PADDING from a client address with a packet number that is the
+largest seen thus far on the connection, the server "commits" to this client
+address. The server MUST send all subsequent packets to this address, with the
+following exception.  PATH_CHALLENGE, PATH_RESPONSE, and PADDING frames are used
+for probing a network path, and receiving only these frames MUST NOT cause the
+server to commit to a new address.
```

Comment:
I think that this can be simpler.  Any packet containing PATH_CHALLENGE is ignored.

You do allow STREAM frames with that packet in earlier text (and STREAM frames aren't a terrible waste of octets, so I can imagine people using the bandwidth for that rather than padding, even if the probability of loss is that much greater).  But this text would cause the recipient to latch.